### PR TITLE
Spring session blog

### PR DESCRIPTION
**Pull request by swethakundaram**

### CODE REVIEW

File: website/blog/2021-01-04-spring-session-persistence.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
+---
+title: "Understanding Scaling in WaveMaker Applications"
+author: Swetha Kundaram
+---
+
+You have developed an exceptional application, and the demand for your application is increasing. You need to expand its accessibility and develop a plan to scale your application. Understanding how to scale an application means ensuring that your application can handle a larger number of users simultaneously.
```
Comment (prasanth-reddy-wavemaker-com) [inline website/blog/2021-01-04-spring-session-persistence.md:6]: we need to add low-code angle to this intro. i will add this statement.

File: website/blog/2021-01-04-spring-session-persistence.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
+---
+title: "Understanding Scaling in WaveMaker Applications"
```
Comment (prasanth-reddy-wavemaker-com) [inline website/blog/2021-01-04-spring-session-persistence.md:2]: replace "in" with "of" Scaling of not Scaling in

File: website/blog/2021-01-04-spring-session-persistence.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+### Horizontal Scaling
+
+Horizontal scaling means it scales an application by adding more machines into the pool across the horizontal direction to "scale-out", and the scaling is based on partitioning the data. It replicates the application by growing the number of [nodes in the cluster](https://www.onixnet.com/insights/kubernetes-101-what-are-nodes-and-clusters#:~:text=Every%20cluster%20has%20one%20master,to%20be%20a%20single%20system.). Also, distributing the responsibilities of each node and providing additional end-points for client connections. 
+
+### Which one is better for your application?
+
+The decision depends on a number of factors, including the following.
+
+1. Is the application request volume steadily growing?
+2. Is the application's current growth experiencing [spikes](https://www.guru99.com/spike-testing.html) that lead to service degradation?
```
Comment (prasanth-reddy-wavemaker-com) [inline website/blog/2021-01-04-spring-session-persistence.md:36]: Need to change the link to guru99 site or remove it completely.

File: website/blog/2021-01-04-spring-session-persistence.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+![Horizontal Scaling Deployment Architecture](/learn/assets/horizontal-scaling.png)
+
+With Horizontal scaling, the data gets distributed among the nodes, and each node contains only a part of the data. Therefore, the user-session created by a node cannot be understood by the other node. In this case, when the user logins again, it rejects the request if the request goes to another node.
+
+Accordingly, with Horizontal scaling, WaveMaker has integrated Spring Session Module to handle session management for the application.
+
+### Integrating Spring Session with WaveMaker
+
+Spring Session manages user-session information while supporting clustered sessions without being tied to an application container specific solution.
```
Comment (prasanth-reddy-wavemaker-com) [inline website/blog/2021-01-04-spring-session-persistence.md:58]: manages `user's http session `

**Review by prasanth-reddy-wavemaker-com — CHANGES_REQUESTED** [review]
Very well written @swethakundaram 💯

**Review by prasanth-reddy-wavemaker-com — APPROVED** [review]

**Review by RameshVarmaKothapalli — APPROVED** [review]

**Review by prasanth-reddy-wavemaker-com — APPROVED** [review]

**Pull request merged**